www/source/blog/2017-08-09-Habitat-plans-for-net-full-iis-apps.md

```diff
@@ -0,0 +1,301 @@
+            PhysicalPath    = Resolve-Path "{{pkg.svc_path}}"
+            ApplicationPool = "{{cfg.app_pool}}"
+            BindingInfo = @(
+                MSFT_xWebBindingInformation
+                {
+                    Protocol = "http"
+                    Port = {{cfg.port}}
+                }
+            )
+        }
+
+        xWebApplication {{cfg.app_name}}
+        {
+            Name = "{{cfg.app_name}}"
+            Website = "{{cfg.site_name}}"
+            WebAppPool =  "{{cfg.app_pool}}"
+            PhysicalPath = Resolve-Path "{{pkg.svc_var_path}}"
+            Ensure = "Present"
+        }
+    }
+}
+```
+
+This uses our Habitat templating capabilities to ensure that IIS points to where our application is stored in the Habitat `svc` folder and uses the configured port.
+
+Our `run` hook will apply this configuration whenever we start our ASP.NET application and make sure everything is configured according to our DSC or `appcmd.exe` calls.
+
+## Running our application in the Habitat supervisor
+
+In full framewoerk apps, IIS is often in charge of the process lifecycle of our app unlike Node or .NET Core where we would simply invoke `node.exe` or `dotnet.exe` to run our app in a light weight web server. So instead of handing off our run hook to call into a runtime we just need to make sure IIS is configured and the site and app pool is in the "running" state. We already covered IIS configuration above. However if the run hook ends, the supervisor assumes our service has terminated. So instead, we will have powershell loop and check that our application endpoint is responsive:
+
+```powershell
+try {
+    Write-Host "{{pkg.name}} is running"
+    $running = $true
+    while($running) {
+        Start-Sleep -Seconds 1
+        $resp = Invoke-WebRequest "http://localhost:{{cfg.port}}/{{cfg.app_name}}" -Method Head
+        if($resp.StatusCode -ne 200) { $running = $false }
+    }
+}
+catch {
+    Write-Host "{{pkg.name}} HEAD check failed"
+}
+finally {
+    # Add any cleanup here which will run after supervisor stops the service
+    Write-Host "{{pkg.name}} is stoping..."
+    ."$env:SystemRoot\System32\inetsrv\appcmd.exe" stop apppool "{{cfg.app_pool}}"
+    ."$env:SystemRoot\System32\inetsrv\appcmd.exe" stop site "{{cfg.site_name}}"
+    Write-Host "{{pkg.name}} has stopped"
+}
+```
+
+As you can see, we simply call `Invoke-Webrequest` to send a `HEAD` request to our app. If the response is not a `200`, then we assume things have gone horribly wrong and exit the loop. Further, if the supervisor stops our service (`hab svc stop mwrock/hab-sln`) execution will move to the `finally` block. Here we tell IIS to shut down our application.
+
+So lets run our application. Assuming we already built it in a studio by entering a Wiondows studio via `hab studio enter -w` from the root of our solution and then running `build ./hab-sln/` to build the hab-sln project. We can stream the supervisor log and start our service:
```

Comment:
"lets" => "let's"

"Wiondows" => "Windows"

"hab-sln" => `hab-sln`

The sentence beginning with "Assuming" isn't complete; did you intend for "We can stream the supervisor..." to be the independent clause in that sentence?